txscript/standard.go

```diff
@@ -132,14 +132,23 @@ func isNullData(pops []parsedOpcode) bool {
 	// OP_RETURN SMALLDATA (where SMALLDATA is a data push up to
 	// MaxDataCarrierSize bytes).
 	l := len(pops)
+
+	// OP_RETURN by itself is allowed.
 	if l == 1 && pops[0].opcode.value == OP_RETURN {
 		return true
 	}
 
+	// Otherwise, OP_RETURN followed by some data is allowed.
+	// This can either be in the form of an OP_PUSHDATA, an
+	// OP_DATA, or an OP_0 through OP_16.
 	return l == 2 &&
 		pops[0].opcode.value == OP_RETURN &&
-		pops[1].opcode.value <= OP_PUSHDATA4 &&
-		len(pops[1].data) <= MaxDataCarrierSize
+		((pops[1].opcode.value <= OP_PUSHDATA4 &&
```

Comment:
I pushed PR #709 which does this via `isSmallInt` which is designed for this purpose and adds a bunch of tests for the various edge conditions.  Please remove this and rebase the PR on top of 709.